File: abc091/b2.c
#include<stdio.h>
#include<string.h>

#define MAX(a, b) a > b ? a : b

int main()
{
    int N,M;
    char s[110][11], t[110][11];
    int i, j, point, ans = 0;

    scanf("%d", &N);

    for (i = 0; i < N; i++)
        scanf("%s", s[i]);

    scanf("%d", &M);

    for (i = 0; i < M; i++)
        scanf("%s", t[i]);

    for (i = 0; i < N; i++)
    {
        point = 0;

        for (j = 0; j < N; j++)
            if (strcmp(s[i], s[j]) == 0) point++;
        for (j = 0; j < M; j++)
            if (strcmp(s[i], t[j]) == 0) point--;

        ans = MAX(point, ans);
    }

    printf("%d", ans);
}

File: abc091/b2.py
N = int(input())
s = [input() for _ in range(0, N)]
M = int(input())
t = [input() for _ in range(0, M)]

ans = 0
for si in range(0, N):
    point = 0
    for i in range(0, N):
        if s[i] == s[si]:
            point += 1
    for i in range(0, M):
        if t[i] == s[si]:
            point -= 1
    ans = max(ans, point)

print(ans)
